pom.xml

```diff
@@ -3186,6 +3186,9 @@
                 <param>com.puppycrawl.tools.checkstyle.checks.coding.SimplifyBooleanExpressionCheckTest</param>
                 <param>com.puppycrawl.tools.checkstyle.checks.coding.SimplifyBooleanReturnCheckTest</param>
               </targetTests>
+              <avoidCallsTo>
+                <avoidCallsTo>java.util.Collections</avoidCallsTo>
```

Comment:
This is too broad exclusion.
Can we narrow to `Collections.unmodifiableSet` ?

Reply:
If there is no way , please prove it .

Absolutely last resort: we can wrap such method in checkstyle util method and exclude by our class.

Reply:
> Can we narrow to Collections.unmodifiableSet ?

no 

> If there is no way , please prove it .

the documentation says that https://pitest.org/quickstart/maven/#:~:text=List%20of%20packages%20and%20classes%20which%20are%20to%20be%20considered%20outside%20the%20scope%20of%20mutation.%20Any%20lines%20of%20code%20containing%20calls%20to%20these%20classes%20will%20not%20be%20mutated. classes will not mutate hence it is not mentioned about a particular method anywhere.
Apart of this i have report only https://kevin222004.github.io/Regression/com.puppycrawl.tools.checkstyle.checks.coding/EqualsAvoidNullCheck.java.html

Reply:
> Absolutely last resort: we can wrap such method in checkstyle util method and exclude by our class.

I am not getting you on this, if we will create and call such kind of method we have to pass the parameter and hence it is not making any difference right know with Collections.unmodifiableSet so it will also not make any impact withour new mwthod as  well it will again a pit failure as a argument propagation one

Reply:
Create class in our project with name `UnmodifableCollectionUtil`
Make method 
```
unmodifiableSet(collection){
 return Collections.unmodifiableSet(collection);
}
```

Add javadoc to this class to explain reason of wrapping of jdk method.
Add our class to pitest config for no mutation.

Reply:
Send feedback to pitest project that it will be awesome extension to allow `avoidCalls` for methods of class. And show this example.

Reply:
Done.

> Send feedback to pitest project that it will be awesome extension to allow avoidCalls for methods of class

I have told them about this in mailing list

Reply:
Please ALWAYS share links to what is done.

Reply:
Share a link please

Reply:
Actually I have posted on mailing list which is not public if they reply then it will visible but i will share on their issue by today

Reply:
That is why better to open ticket on GitHub. In worse case they will close it won't fix. It gives all others from world to participate and support idea

Reply:
> That is why better to open ticket on GitHub. In worse case they will close it won't fix. It gives all others from world to participate and support idea

Yes, from next time will take care of this 

I have opened an issue https://github.com/hcoles/pitest/issues/1233